m/processing_tools/gridGliderData.m

```diff
@@ -306,26 +306,31 @@
   fprintf('  number of variables   : %d\n', num_variables);
   data_grid_variables = nan(num_casts, num_levels, num_variables);
   for cast_idx = 1:num_casts
-    cast_select = (profile == cast_idx);
-    cast_lat = latitude(cast_select);
-    cast_lon = longitude(cast_select);
-    cast_depth = depth(cast_select);
-    cast_time = time(cast_select);
-    cast_variables = variables(cast_select, :);
-    data_grid.time(cast_idx) = nanmean(cast_time);
-    data_grid.latitude(cast_idx) = nanmean(cast_lat);
-    data_grid.longitude(cast_idx) = nanmean(cast_lon);
-    if ~isempty(cast_variables) % Speed up when there are no variables.
-      data_grid_variables(cast_idx, :, :) = ...
-        cell2mat(arrayfun(@(d) nanmean(cast_variables(abs(cast_depth-d)<=0.5*depth_resolution, :), 1), ...
-                          depth_range(:), 'UniformOutput', false));
-    end
-  end
-  % Move binned variable data to output struct.
-  for variable_name_idx = 1:num_variables
-    variable_name = variable_name_list{variable_name_idx};
-    data_grid.(variable_name) = data_grid_variables(:, :, variable_name_idx);
-  end
+      cast_select = (profile == cast_idx);
+      cast_lat = latitude(cast_select);
+      cast_lon = longitude(cast_select);
+      cast_depth = depth(cast_select);
+      cast_time = time(cast_select);
+      cast_variables = variables(cast_select, :);
+      data_grid.time(cast_idx) = nanmean(cast_time);
+      data_grid.latitude(cast_idx) = nanmean(cast_lat);
+      data_grid.longitude(cast_idx) = nanmean(cast_lon);
+
+      % Speed up when there are no variables. % -FC modifs
+      if ~isempty(cast_variables)
+          cast_mean = arrayfun(@(d) nanmean(cast_variables(abs(cast_depth-d)<=0.5*depth_resolution,:),1), depth_range(:), 'UniformOutput', false);
+          
+          cast_good_idx = cellfun(@(d) any(~isnan(d(:))), cast_mean);     
+          cast_mean = cell2mat(cast_mean(cast_good_idx));
+          data_grid_variables(cast_idx, cast_good_idx, :) =  cast_mean;
+      end
+   end
+  
+   % Move binned variable data to output struct.
+   for variable_name_idx = 1:num_variables
+       variable_name = variable_name_list{variable_name_idx};
+       data_grid.(variable_name) = data_grid_variables(:, :, variable_name_idx);
+   end
```

Comment:
Matlab 7.11.0.584 (R2010b)

Same output as you except last one:
`>> nanmean(zeros(0,3), 1)`
`ans =`
 `     NaN`

Are you using NanSuite from file exchange?